test/unittest/dds/xtypes/dyndata_idl/idl_to_json_Tests.cpp

```diff
@@ -0,0 +1,443 @@
+                ]
+            ]
+        ],
+        "my_aliased_bounded_string": "",
+        "my_aliased_enum": {
+            "name": "A",
+            "value": 0
+        },
+        "my_aliased_struct": {
+            "my_bool": false,
+            "my_char": "\u0000",
+            "my_double": 0.0,
+            "my_float": 0.0,
+            "my_int8": 0,
+            "my_long": 0,
+            "my_longdouble": 0.0,
+            "my_longlong": 0,
+            "my_octet": 0,
+            "my_short": 0,
+            "my_uint8": 0,
+            "my_ulong": 0,
+            "my_ulonglong": 0,
+            "my_ushort": 0,
+            "my_wchar": "\u0000"
+        },
+        "my_bitmask": {
+            "active": [],
+            "binary": "00000000000000000000000000000000",
+            "value": 0
+        },
+        "my_bitset": {
+            "a": 0,
+            "b": false,
+            "c": 0,
+            "d": 0
+        },
+        "my_bool": false,
+        "my_bounded_string": "",
+        "my_bounded_wstring": "",
+        "my_char": "\u0000",
+        "my_double": 0.0,
+        "my_enum": {
+            "name": "A",
+            "value": 0
+        },
+        "my_float": 0.0,
+        "my_int8": 0,
+        "my_long": 0,
+        "my_longdouble": 0.0,
+        "my_longlong": 0,
+        "my_octet": 0,
+        "my_recursive_alias": {
+            "name": "A",
+            "value": 0
+        },
+        "my_short": 0,
+        "my_string": "",
+        "my_uint8": 0,
+        "my_ulong": 0,
+        "my_ulonglong": 0,
+        "my_ushort": 0,
+        "my_wchar": "\u0000",
+        "my_wstring": "",
+        "short_long_map": null,
+        "short_sequence": [],
+        "string_alias_unbounded_map": null,
+        "string_unbounded_map": null
+    }
+})";
+
+TEST(idl_to_json_Tests, DataTest)
+{
+    // std::string topic_name = "TypeIntrospectionTopic";
+    DataTypeKind data_type_complete = DataTypeKind::DATA_TEST;
+
+    auto data_type_complete_ = data_type_factory(data_type_complete);
+
+    eprosima::fastdds::dds::DynamicType::_ref_type dynamic_type_complete = data_type_complete_->generate_type_support_and_get_dyn_type();
+
+    // Get dynamic data depending on data_type
+    auto dyn_data_complete = eprosima::fastdds::dds::DynamicDataFactory::get_instance()->create_data(dynamic_type_complete);
+
+    dyn_data_complete->set_uint32_value(dyn_data_complete->get_member_id_by_name("index"), 3);
+
+    std::stringstream generated_json_complete;
+    generated_json_complete << std::setw(4);
+    if (eprosima::fastdds::dds::RETCODE_OK ==
+            eprosima::fastdds::dds::json_serialize(
+                dyn_data_complete,
+                generated_json_complete,
+                eprosima::fastdds::dds::DynamicDataJsonFormat::EPROSIMA))
+    {
+        std::cout << "---------------" << std::endl;
+        std::cout << generated_json_complete.str() << std::endl;
+        std::cout << "---------------" << std::endl;
```

Comment:
I'm not sure about Fast-DDS standards, but maybe it would better not to print this.